### DISCUSSION TITLE
Combine (almost) all individual image props into a single imgAttributes prop

### DISCUSSION THREAD

**Discussion by laurenashpole** [laurenashpole/react-inner-image-zoom#77] [Ideas]
@viniciuscasarin Sorry for the delay starting this discussion! The last couple weeks were busier than expected. If you're still interested and have the time, here's what I was thinking for the next major version update:

Right now the `srcSet`, `sizes`,  `alt`, and `title` props are just passed directly to the `img` tag. Instead of handling them individually, we could switch to accepting an object called `imgAttributes` as a prop which would allow users to include any img or global attributes. I left out `src`, `width`, and `height` because a) it seems like the src prop should still be required and b) width and height are used in other places to create the loading spacer.

As part of this, it might also make sense to update the `source` tag to pass in props using a spread operator instead of individual attributes to make that more flexible too.

I'd love to hear what you (or anyone else who wants to jump in) think. And no pressure if you don't have the time to work on this right now, just let me know and I can take a look.

**Comment by viniciuscasarin** [comment]
It sounds great, ill start it right away

**Reply by laurenashpole** [reply]
Hey @viniciuscasarin! Just wanted to check in on this. I was thinking of doing a quick round of bug fixes and it would be nice to include. I know things can get hectic this time of year (and really have felt kind of hectic the past two years) so let me know if you need me to take it off your plate. Thanks!

**Reply by viniciuscasarin** [reply]
Hey @laurenashpole 
I didnt get the chance to focus in this implementation as much as I was hoping to. Its fine by me if you want to work on it :)
Thanks for your support!

**Reply by laurenashpole** [reply]
Totally understand. I just happen to have some free time so I might take a look. If you see anything else come in that you want to grab let me know.
